### PR TITLE
Cli: flip no_extend to auto_extend internally

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
https://github.com/anza-xyz/agave/pull/791 adds some unintuitive double-negative logic to the CLI (eg `!no_extend`)

#### Summary of Changes
Change internal logic: no_extend > auto_extend

### CODE REVIEW

File: cli/src/program.rs [hunk 1]
```diff
@@ -276,7 +276,7 @@ impl ProgramSubCommands for App<'_, '_> {
                         ))
                         .arg(compute_unit_price_arg())
                         .arg(
-                            Arg::with_name("no_extend")
+                            Arg::with_name("no_auto_extend")
```
Comment (buffalojoec) [inline cli/src/program.rs:279]: nit: The long arg should match, ie. `no-auto-extend`. (Below this line)

File: cli/tests/program.rs [hunk 2]
```diff
@@ -824,7 +824,7 @@ fn test_cli_program_upgrade_auto_extend() {
         skip_fee_check: false,
         compute_unit_price: None,
         max_sign_attempts: 5,
-        no_extend: true, // --no-extend (true)
+        auto_extend: false, // --no-extend present (true)
```
Comment (buffalojoec) [inline cli/tests/program.rs]: tiny nit on the comment:

```suggestion
        auto_extend: false, // --no-auto-extend flag is present (true)
```

File: cli/tests/program.rs [hunk 3]
```diff
@@ -846,7 +846,7 @@ fn test_cli_program_upgrade_auto_extend() {
         skip_fee_check: false,
         compute_unit_price: None,
         max_sign_attempts: 5,
-        no_extend: false, // --no-extend (false)
+        auto_extend: true, // --no-extend not present (false)
```
Comment (buffalojoec) [inline cli/tests/program.rs]: Same tiny nit here!

**Review by buffalojoec — COMMENTED** [review]
hehe, "here, I'll just show you what I mean".

I like it, thank you!

**Comment by CriesofCarrots** [comment]
> hehe, "here, I'll just show you what I mean".

Sometimes it's easier 😅

File: cli/src/program.rs (hunk 1 above)
Comment (CriesofCarrots) [reply cli/src/program.rs:279]: Yep, sure, I like the longer arg better anyway. Just didn't want to whiplash if people were set on the existing flag.

File: cli/tests/program.rs (hunk 3 above)
Comment (CriesofCarrots) [reply cli/tests/program.rs]:
```suggestion
        auto_extend: true, // --no-auto-extend flag is not present (false)
```

File: cli/src/program.rs [hunk 4]
```diff
@@ -276,7 +276,7 @@ impl ProgramSubCommands for App<'_, '_> {
...
                                 .long("no-extend")
```
Comment (CriesofCarrots) [inline cli/src/program.rs]:
```suggestion
                                .long("no-auto-extend")
```

**Review by buffalojoec — APPROVED** [review]
Thanks again!

**Pull request merged**